src/systems/sprites.js

```diff
@@ -211,81 +222,104 @@ export class SpriteSystem {
-      this.mesh.frustumCulled = false;
-      this.mesh.renderOrder = window.APP.RENDER_ORDER.HUD_ICONS;
-      this.mesh.raycast = this.raycast.bind(this);
+
+    const domReady = waitForDOMContentLoaded();
+
+    Promise.all([domReady]).then(() => {
+      for (const [spritesheetPng, type] of [[spritesheetActionPng, "action"], [spritesheetNoticePng, "notice"]]) {
+        Promise.all([createImageTexture(spritesheetPng, getThemeColorShifter(type)), waitForDOMContentLoaded()]).then(
+          ([spritesheetTexture]) => {
+            const material = new THREE.RawShaderMaterial({
+              uniforms: {
+                u_spritesheet: { value: spritesheetTexture },
+                hubs_Time: { value: 0 }
+              },
+              vertexShader,
+              fragmentShader,
+              side: THREE.DoubleSide,
+              transparent: true
+            });
+            const mesh = (this.meshes[type] = new THREE.Mesh(createGeometry(this.maxSprites), material));
+            const el = document.createElement("a-entity");
+            el.classList.add("ui");
+            scene.appendChild(el);
+            el.setObject3D("mesh", mesh);
+            mesh.frustumCulled = false;
+            mesh.renderOrder = window.APP.RENDER_ORDER.HUD_ICONS;
+            mesh.raycast = this.raycast.bind(this);
+          }
+        );
+      }
     });
   }
 
   tick(t) {
-    if (!this.mesh) return;
-
-    this.mesh.material.uniforms.hubs_Time.value = t;
-    this.mesh.material.uniformsNeedUpdate = true;
-
-    const mvCols = this.mesh.geometry.attributes["mvCol0"].data; // interleaved
-    const aEnableSweepingEffect = this.mesh.geometry.attributes["a_hubs_EnableSweepingEffect"];
-    const aSweepParams = this.mesh.geometry.attributes["a_hubs_SweepParams"];
-    for (let i = 0; i < this.slots.length; i++) {
-      if (!this.slots[i]) continue;
-
-      const sprite = this.indexWithSprite.get(i);
-      if (isVisible(sprite.el.object3D)) {
-        const enableSweepingEffectValue = enableSweepingEffect(sprite) ? 1 : 0;
-        aEnableSweepingEffect.array[i * 4 + 0] = enableSweepingEffectValue;
-        aEnableSweepingEffect.array[i * 4 + 1] = enableSweepingEffectValue;
-        aEnableSweepingEffect.array[i * 4 + 2] = enableSweepingEffectValue;
-        aEnableSweepingEffect.array[i * 4 + 3] = enableSweepingEffectValue;
-        aEnableSweepingEffect.needsUpdate = true;
-        if (enableSweepingEffectValue) {
-          const hoverableVisuals = getHoverableVisuals(sprite.el);
-          const s = hoverableVisuals.sweepParams[0];
-          const t = hoverableVisuals.sweepParams[1];
-          aSweepParams.setXY(4 * i + 0, s, t);
-          aSweepParams.setXY(4 * i + 1, s, t);
-          aSweepParams.setXY(4 * i + 2, s, t);
-          aSweepParams.setXY(4 * i + 3, s, t);
-          aSweepParams.needsUpdate = true;
+    if (!this.meshes.action || !this.meshes.notice) return;
+
+    for (const [sheetType, mesh] of Object.entries(this.meshes)) {
```

Comment:
`Object.entries` allocates a new array. Probably shouldn't do it in `tick` or `raycast`